src/server/game/Entities/Pet/Pet.h

```diff
@@ -167,6 +169,7 @@ class TC_GAME_API Pet : public Guardian
 
         DeclinedName *m_declinedname;
 
+        uint8 m_specGroup;
```

Comment:
You don't need to store it - you can find it in ChrSpecializationEntry::OrderIndex (and you can get ChrSpecializationEntry by a simple dbc lookup with m_petSpecialization)